corehq/apps/case_search/xpath_functions/comparison.py

```diff
@@ -50,8 +59,31 @@ def property_comparison_query(context, case_property_name_raw, op, value_raw, no
             )
 
 
+def _create_timezone_adjusted_datetime_query(case_property_name, op, value, node, timezone):
+    """
+    Given a date, it gets the equivalent starting time of that date in UTC. i.e 2023-06-05
+    in Asia/Seoul timezone begins at 2023-06-04T20:00:00 UTC.
+    This might be inconsistent in daylight savings situations.
+    """
+    utc_equivalent_datetime_value = adjust_input_date_by_timezone(value_to_date(node, value), timezone, op)
+    if op in [EQ, NEQ]:
+        day_start_datetime = utc_equivalent_datetime_value.isoformat()
+        day_end_datetime = (utc_equivalent_datetime_value + timedelta(days=1)).isoformat()
+        op_value_dict = {
+            RANGE_OP_MAPPING[">="]: day_start_datetime,
+            RANGE_OP_MAPPING["<="]: day_end_datetime,
```

Comment:
I think this should be `<` so nothing can fall on two dates.

Reply:
yes you're right.

Going to address your comments here https://github.com/dimagi/commcare-hq/pull/33286 since this PR is already merged.